src/hotspot/share/runtime/continuationFreezeThaw.cpp

```diff
@@ -2154,14 +2150,12 @@ NOINLINE void ThawBase::recurse_thaw_interpreted_frame(const frame& hf, frame& c
 
   assert((stack_frame_bottom >= stack_frame_top + fsize) &&
          (stack_frame_bottom <= stack_frame_top + fsize + 1), ""); // internal alignment on aarch64
-
-  // on AArch64/PPC64 we add padding between the locals and the rest of the frame to keep the fp 16-byte-aligned
-  const int locals = hf.interpreter_frame_method()->max_locals();
   assert(hf.is_heap_frame(), "should be");
   assert(!f.is_heap_frame(), "should not be");
 
-  copy_from_chunk(heap_frame_bottom - locals, stack_frame_bottom - locals, locals); // copy locals
-  copy_from_chunk(heap_frame_top, stack_frame_top, fsize - locals);                 // copy rest
+  // Some architectures (like AArch64/PPC64/RISC-V) adds padding between the locals and the fixed_frame to keep the fp 16-byte-aligned.
+  // On those architectures we thaw the padding in order to keep the same localized pointer values.
```

Comment:
Somehow this suggestion was not delivered...
```suggestion
  // On those architectures we thaw the padding in order to keep the same relative references.
```

Reply:
This is very unclear. I think it means the same "relative pointers," otherwise known as "offsets."

Reply:
Tried to clarify by changing the comment to:
`// On those architectures we freeze the padding in order to keep the same fp-relative offsets in the fixed_frame.`